koku/api/iam/customer_manager.py

```diff
@@ -59,10 +60,26 @@ def __init__(self, uuid):
         except ValidationError as e:
             raise(CustomerManagerError(str(e)))
 
+        try:
+            self.tenant = self._get_tenant_query().get()
+        except Tenant.DoesNotExist as e:
+            msg = 'No tenant found for {}'.format(self.get_name())
+            LOG.error(msg)
+
     def get_model(self):
         """Get the model object for the customer."""
         return self.model
 
+    def _get_tenant_query(self):
+        """Get the tenant model query for the customer."""
+        customer_schema_name = self.get_model().schema_name
+        query = Tenant.objects.all().filter(schema_name=customer_schema_name)
```

Comment:
I'm pretty sure you can simplify this from `Tenant.objects.all().filter(schema_name=customer_schema_name)` to `Tenant.objects.filter(schema_name=customer_schema_name)`

Similar statement for other filter calls here. No need to get all records first.